packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -164,10 +171,13 @@ export class TeamsBotV2Impl {
   private getTemplates(ctx: Context, inputs: Inputs): CodeTemplateInfo[] {
     const actRoles = this.resolveActRoles(ctx, inputs);
     const triggers = this.resolveTriggers(inputs);
-    const hostType = this.resolveHostType(ctx);
+    const hostType = ctx.projectSetting?.pluginSettings?.[PluginBot.PLUGIN_NAME]?.[
+      PluginBot.HOST_TYPE
+    ] as HostType;
     const lang = this.resolveProgrammingLanguage(ctx);
+    const isM365 = ctx.projectSetting?.isM365;
 
-    const scenarios = this.resolveScenarios(actRoles, triggers, hostType);
+    const scenarios = this.resolveScenarios(actRoles, triggers, hostType, isM365);
 
     return scenarios.map((scenario) => {
       return {
```

Comment:
Add comment for getBicepConfigs() in L192, since there is only one type of programming language, the line between 197 and 202 is a relation of if...else. The value can be decided by lang and it could be a map, instead of multiple if conditions.

Reply:
Change it to use map here. Please check.